File: calendarlibrary/src/main/java/com/sunyuan/calendarlibrary/model/CalendarDay.java
package com.sunyuan.calendarlibrary.model;

import java.io.Serializable;
import java.util.Calendar;
import java.util.Date;
import java.util.Objects;

/**
 * author：six
 * created by:2019-01-20
 * github:https://github.com/sy007
 */
public class CalendarDay implements Serializable {

    private int year;
    private int month;
    private int day;

    public CalendarDay(int year, int month, int day) {
        this.year = year;
        this.month = month;
        this.day = day;
    }


    public Date toDate() {
        Calendar calendar = Calendar.getInstance();
        calendar.set(Calendar.YEAR, year);
        calendar.set(Calendar.MONTH, month);
        calendar.set(Calendar.DATE, day);
        return calendar.getTime();
    }

    public int getYear() {
        return year;
    }

    public void setYear(int year) {
        this.year = year;
    }

    public int getMonth() {
        return month;
    }

    public void setMonth(int month) {
        this.month = month;
    }

    public int getDay() {
        return day;
    }

    public void setDay(int day) {
        this.day = day;
    }


    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof CalendarDay)) return false;
        CalendarDay that = (CalendarDay) o;
        return getYear() == that.getYear() &&
                getMonth() == that.getMonth();
    }

    @Override
    public int hashCode() {
        return Objects.hash(getYear(), getMonth());
    }

    @Override
    public String toString() {
        return "CalendarBean{" +
                "year=" + year +
                ", month=" + month +
                ", day=" + day +
                '}';
    }
}

File: calendar-view/src/main/java/com/sy007/calendar/entity/CalendarDay.kt
package com.sy007.calendar.entity

import com.sy007.calendar.before
import java.io.Serializable
import java.util.*

/**
 * 日期模型类
 * @author sy007
 * @date 2019/01/20
 */
class CalendarDay : Serializable, Comparable<CalendarDay> {

    private val tempCalendar = Calendar.getInstance()

    /**
     * 当前绘制的年份
     */
    var year: Int

    /**
     * 当前绘制的月份，从0-11
     */
    var month: Int

    /**
     * 当前绘制的天，从1-31
     */
    var day: Int

    /**
     * 当前时间年份
     */
    val curYear: Int = tempCalendar[Calendar.YEAR]

    /**
     * 当前时间月份，从0-11
     */
    val curMonth: Int = tempCalendar[Calendar.MONTH]

    /**
     * 当前时间的日子，从1-31
     */
    val curDay: Int = tempCalendar[Calendar.DAY_OF_MONTH]

    /**
     * 当前日期属于哪一月，默认为当前月
     */
    var dayOwner = DayOwner.THIS_MONTH


    /**
     * 当前日期属于哪一月
     */
    enum class DayOwner {
        /**
         * 当前月的前一月
         */
        PRE_MONTH,

        /**
         * 当前月
         */
        THIS_MONTH,

        /**
         * 当前月的后一月
         */
        NEXT_MONTH
    }


    constructor(year: Int, month: Int, day: Int) {
        this.year = year
        this.month = month
        this.day = day
    }

    constructor(calendar: Calendar) : this(calendar[Calendar.YEAR], calendar[Calendar.MONTH], calendar[Calendar.DAY_OF_MONTH])


    /**
     * 当前日期的Calendar
     */
    val calendar: Calendar
        get() {
            tempCalendar.clear()
            tempCalendar[Calendar.YEAR] = year
            tempCalendar[Calendar.MONTH] = month
            tempCalendar[Calendar.DAY_OF_MONTH] = day
            return tempCalendar
        }

    /**
     * 当前时间的Calendar
     */
    val curCalendar: Calendar
        get() {
            tempCalendar.clear()
            tempCalendar[Calendar.YEAR] = curYear
            tempCalendar[Calendar.MONTH] = curMonth
            tempCalendar[Calendar.DAY_OF_MONTH] = curDay
            return tempCalendar
        }

    override fun toString(): String {
        return "CalendarDay{" +
                "year=" + year +
                ", month=" + month +
                ", day=" + day +
                '}'
    }


    override fun compareTo(other: CalendarDay): Int {
        if (equals(other)) {
            return 0
        }
        return if (before(other)) -1 else 1
    }

    override fun equals(other: Any?): Boolean {
        val that = other as? CalendarDay ?: return false
        if (year != that.year) return false
        if (month != that.month) return false
        if (day != that.day) return false
        if (dayOwner != that.dayOwner) return false
        return true
    }

    override fun hashCode(): Int {
        var result = year
        result = 31 * result + month
        result = 31 * result + day
        result = 31 * result + dayOwner.hashCode()
        return result
    }
}